server/service/src/sync/translation_and_integration.rs

```diff
@@ -138,27 +138,27 @@ impl IntegrationRecords {
         // Only start nested transaction if transaction is already ongoing. See integrate_and_translate_sync_buffer
         let start_nested_transaction = { connection.transaction_level.get() > 0 };
 
-        for upsert in self.upserts.iter() {
+        for delete in self.deletes.iter() {
```

Comment:
Swapped deletes with upserts since program requisitions use both in the same cycle. Have ran integration tests and they still pass.